src/video.js

```diff
@@ -24,7 +24,7 @@ async function getVideoDimensions (urlOrFilename) {
   const stream = metadata.streams.find(s => s.codec_type === 'video');
   if (!stream) throw new Error('video stream not found');
   const rotation = getVideoRotation(stream);
-  const { width, height, duration } = stream;
+  const { width, height, duration = metadata.format.duration } = stream;
```

Comment:
I'm assuming `metadata.format` always exists, is an object and won't throw any duration of undefined errors?

Reply:
I'll change this to be `duration = (metadata.format && metadata.format.duration || 0)` in case it isn't an object